### PR TITLE
Add possibility to focus on search input using keyboard

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
This PR adds the possibility to focus on the search input with `S` or `/` like in rustdoc and `mdbook` and `docs.rs` (unification++). Pressing escape will blur it.

r? @Alexendoo 

changelog: Add possibility to focus on search input using keyboard

### CODE REVIEW

File: util/gh-pages/script.js [hunk 1]
```diff
@@ -579,6 +579,40 @@ function setTheme(theme, store) {
...
+    if ("key" in ev && typeof ev.key !== "undefined") {
+        return ev.key;
+    }
+
+    const c = ev.charCode || ev.keyCode;
+    if (c === 27) {
+        return "Escape";
+    }
+    return String.fromCharCode(c);
+}
```
Comment (Alexendoo) [inline util/gh-pages/script.js:592]: I don't think we need the fallback looking at the [browser support](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key#browser_compatibility)

File: util/gh-pages/script.js [hunk 2]
```diff
@@ -579,6 +579,40 @@ function setTheme(theme, store) {
...
+    if (document.activeElement.tagName === "INPUT") {
+        if (getVirtualKey(ev) === "Escape") {
+            document.activeElement.blur();
+        }
+    } else {
+        switch (getVirtualKey(ev)) {
+            case "s":
+            case "S":
+            case "/":
+                ev.preventDefault(); // To prevent the kep to be put into the input.
```
Comment (Alexendoo) [inline util/gh-pages/script.js:604]: typo - key

Comment (Alexendoo) [reply util/gh-pages/script.js:604]: We probably should add an early return here also if there's a modifier key on, it doesn't seem super likely but someone could want to `ctrl + s` to save the page

Comment (GuillaumeGomez) [reply util/gh-pages/script.js:604]: Very good point! Adding that.

File: util/gh-pages/script.js (hunk 1 above)
Comment (GuillaumeGomez) [reply util/gh-pages/script.js:592]: Fair enough. Removing it.

**Comment by GuillaumeGomez** [comment]
Updated!

File: util/gh-pages/script.js [hunk 3]
```diff
@@ -579,6 +579,32 @@ function setTheme(theme, store) {
     }
 }
 
+function handleShortcut(ev) {
+    if (ev.ctrlKey || ev.altKey || ev.metaKey || disableShortcuts) {
```
Comment (Alexendoo) [inline util/gh-pages/script.js:583]: Where's `disableShortcuts` from? I don't see it in the rest of the script

Comment (GuillaumeGomez) [reply util/gh-pages/script.js]: From my tired brain. ^^'

Removed it!

**Review by Alexendoo — APPROVED** [review]

**Comment by Alexendoo** [comment]
Thanks! This will be a nice QoL bump

@bors r+

**Pull request merged**